io/io/src/TBufferJSON.cxx

```diff
@@ -1610,61 +1610,82 @@ void TBufferJSON::JsonWriteCollection(TCollection *col, const TClass *)
    // collection treated as JS Array
    AppendOutput("[");
 
-   bool islist = col->InheritsFrom(TList::Class());
    TMap *map = nullptr;
-   if (col->InheritsFrom(TMap::Class()))
+   TList *lst = nullptr;
+   if (col->InheritsFrom(TList::Class()))
+      lst = dynamic_cast<TList *>(col);
+   else if (col->InheritsFrom(TMap::Class()))
```

Comment:
Apriori we can simply the code with:
```suggestion
   TList *lst = dynamic_cast<TList *>(col);
   TMap *map = dynamic_cast<TMap*>(col);
```
i.e. I don't think that `InheritsFrom` is much faster than the `dynamic_cast`.

Reply:
I add commit with such changes